### DISCUSSION TITLE
To discuss: switch from SMILES to CXSMILES as central structural data representation

### DISCUSSION THREAD

**Discussion by JonasSchaub** [FelixBaensch/MORTAR#126] [Ideas]
I am guessing that this would not require much more RAM but could allow us to represent more structural features. But we need to check whether all these features can actually be handled by MORTAR or will cause issues.

**Comment by FelixBaensch** [comment]
Does CDK support CXSMILES natively?

**Comment by JonasSchaub** [comment]
From the [CxSmilesParser class](https://github.com/cdk/cdk/blob/main/storage/smiles/src/main/java/org/openscience/cdk/smiles/CxSmilesParser.java) documentation:

```
/**
 * Parse CXSMILES (ChemAxon Extended SMILES) layers. The layers are suffixed after the SMILES but before the title
 * and encode a large number of the features. CXSMILES was not intended for outside consumption so has some quirks
 * but does provide some useful features. This parser handles a subset of the grammar:
 * <br>
 * <pre>
 * - Atom Labels
 * - Atom Values
 * - Atom Coordinates
 * - Positional Variations
 * - Polymer Sgroups
 * - Atom Radicals
 * - Fragment grouping
 * </pre>
 * The following properties are ignored
 * <pre>
 * - cis/trans specification
 * - relative stereochemistry
 * </pre>
 */
```

At least for atom coordinates, if given in the input, it might be helpful. But maybe keep this as an idea for now and see how the CDK support develops.
